src/controllers/midi/midimessage.h

```diff
@@ -178,40 +181,69 @@ struct MidiKey {
     };
 };
 
+template<class>
+inline constexpr bool always_false_v = false;
+
 struct MidiInputMapping {
-    MidiInputMapping() {
+    MidiInputMapping()
+            : uuid(QUuid::createUuid()) {
     }
 
     MidiInputMapping(MidiKey key, MidiOptions options)
             : key(key),
-              options(options) {
+              options(options),
+              uuid(QUuid::createUuid()) {
     }
 
-    MidiInputMapping(MidiKey key, MidiOptions options, const ConfigKey& control)
+    MidiInputMapping(MidiKey key,
+            MidiOptions options,
+            const std::variant<ConfigKey, QJSValue>& control)
             : key(key),
               options(options),
-              control(control) {
+              control(control),
+              uuid(QUuid::createUuid()) {
```

Comment:
is the Uuid really required? Is relying on QJSValue::strictlyEquals too weak for comparision?